### DISCUSSION TITLE
pag注册解码器不生效

### DISCUSSION THREAD

**Discussion by hsyfee** [Tencent/libpag#1946] [Q&A]
**版本：**

- "ffavc": "^1.0.2",
- "libpag": "^4.2.84",
- "react": "18.2.0",

**环境：**
nextjs h5

**问题**
想解决h5部分浏览器pag动效不渲染或者渲染但是没有动效的问题，
按照官方文档注册解码器，浏览器一直弹出下图警告，几千条，并且浏览器会卡死。
如果注释掉PAG.registerSoftwareDecoderFactory(ffavcDecoderFactory);则正常渲染

`FFAVCInit({ locateFile: (file: string) => `${prefix}/${file}` }).then(
      (FFAVC) => {
        PAGInit({
          locateFile: (file) => `${prefix}/${file}`,
        }).then((PAG) => {
          // 注册解码器
          const ffavcDecoderFactory = new FFAVC.FFAVCDecoderFactory();
          PAG.registerSoftwareDecoderFactory(ffavcDecoderFactory);

          fetch(newPagUrl)
            .then((res) => res.arrayBuffer())
            .then(async (buffer) => {
              const pagFile = await PAG.PAGFile.load(buffer);
              const canvas = document.getElementById(
                canvasId
              ) as HTMLCanvasElement;
              canvas.width = pagFile.width();
              canvas.height = pagFile.height();
              const pagView = await PAG.PAGView.init(pagFile, canvas);
              pagView?.setRepeatCount(0);
              await pagView?.play();
            });
        });
      }
    `);`
    
    
![warning](https://github.com/Tencent/libpag/assets/80185494/c399f079-5e6e-4b87-81b3-a88390426e4d)

**Comment by CodeJhF** [comment]
抱歉这条当时没有被跟进到，是我们漏了。

因长期无回复，且 libpag 4.3 及更早版本已停止维护（当前仅维护 4.4 / 4.5），先关闭本条。
上报版本：1.0.2 / 4.2.84。

如果问题在最新版本（4.4 / 4.5）仍能复现，欢迎在本帖回复或新开 issue，并附上 libpag 版本号、平台与系统版本、可复现的 PAG 文件或最小 demo，我们会优先跟进。（stale-close）
